src/Sylius/Bundle/TranslationBundle/DependencyInjection/Compiler/TranslationListenerPass.php

```diff
@@ -24,19 +25,38 @@ class TranslationListenerPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('sylius.translatable.listener')) {
-            return;
-        }
+        $driver = $container->getParameter('sylius_translation.driver');
```

Comment:
You are right, but still this can be simplified. I.e. you can't use both drivers so you can register definition on choosen value of config in extension instead of registering both in `service.xml`.